### PR TITLE
Windows debugging

### PR DESCRIPTION
**Pull request by brooksmtownsend**
This PR was created in an attempt to further debug the Windows issues that we are running into for wasmcloud.

TL;DR: use WSL

Re: #60, the libloading library automatically appends `.dll` which I confirmed in their documentation, so we have to extract the provider with the `.dll` extension or windows machines will not be able to work.

Refactored the libloading section to have a linux, windows, and "other" section for loading capability providers. I was able to discover that the `.dll` section is not required so long as you use the `libloading::os::windows` module for the library loading portion. I was not able to resolve https://github.com/wasmCloud/wasmCloud/issues/59, and windows still can segfault upon initialization of the capability provider. I was able to confirm with `dumpbin` that the downloaded httpserver provider (I'm running the echo example) does export `__capability_provider_create`, but regardless the code panics with a `LoadLibraryExW` error. From what I can read, this is due to missing shared libraries elsewhere on my system, but as far as I know I have a well functioning Windows development environment. 

I've used `depends` to try and track down the shared module that is failing, however my `depends` lists hundreds of missing `.dll`s.

I've used ProcessMonitor to track syscalls for the wasmcloud program, but I can't find anything out of the ordinary other than a failure to `CreateFileMapping`. I've attempted to open the library with the specified protections, but unfortunately this doesn't fix the issue. 

Here's my current error:
```
[2021-03-30T18:14:31Z DEBUG] GetProcAddress { source: Os { code: 126, kind: Other, message: "The specified module could not be found." } }
[2021-03-30T18:14:31Z ERROR] Failed to extract plugin from provider: GetProcAddress failed
```

Interested if anyone else on windows can reproduce, if you have `nats-server` and I stole the manifest for the program from https://github.com/wasmCloud/examples/blob/main/echo/manifest.yaml

Reference:
https://docs.microsoft.com/en-us/windows/win32/memory/memory-protection-constants
https://docs.rs/libloading/0.7.0/libloading/os/windows/struct.Library.html

### CODE REVIEW

**Comment by autodidaddict** [comment]
w/r/t missing shared files - have you installed a windows equivalent of the C runtime? On linux we need that for libloading to work.

**Comment by brooksmtownsend** [comment]
@autodidaddict Unfortunately I've spent hours trying to make sure my runtime environment is correct, and I'm like 90% sure that my configuration is the correct one. Originally I started on `x86_64-pc-windows-gnu`, only to find out that the recommended way is to install visual studio (not code) and get all of the build tools through there. Which I was able to do, and I even use the whole cross tools command prompt, but its still possible I don't have the proper toolchain installed. If there's any easy resource to check that I'm happy to try.

**Comment by autodidaddict** [comment]
Thanks for the diligence on this. I think you're right in that, at least for now, we recommend people use WSL when working on windows.

**Comment by brooksmtownsend** [comment]
> Thanks for the diligence on this. I think you're right in that, at least for now, we recommend people use WSL when working on windows.

This does narrow our scope, but as you mentioned (potentially in slack) this only really affects users that intend to use windows in production as WSL is a small change for a development machine. We'll just have to take this stance in our documentation somewhere as far as OS/platforms that we officially support

**Comment by brooksmtownsend** [comment]
Since https://github.com/wasmCloud/wasmCloud/issues/60 is closed, closing this PR. Picking up any related docs in https://github.com/wasmCloud/wasmcloud-dev-site/issues/28
